src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseDataFileService.cs

```diff
@@ -452,13 +449,19 @@ private async Task<DataFileInfo> GetFallbackDataFileInfo(Guid releaseId, File da
 
         private async Task<string> GetSubjectName(File file)
         {
-            if (file.SubjectId.HasValue)
+            if (file.Type != FileType.Data)
             {
-                var subject = await _statisticsDbContext.Subject.FindAsync(file.SubjectId.Value);
-                return subject.Name;
+                throw new ArgumentException("file.Type should equal FileType.Data");
             }
 
-            return "Unknown";
+            var releaseFile = await _contentDbContext
+                .ReleaseFiles
+                .Include(rf => rf.File)
+                .SingleOrDefaultAsync(rf =>
```

Comment:
This is going to blow up when the file is reused on an amendment because you aren't constraining it by Release Id here. Also once you've got the releaseId to this method you can just do something along the lines of  `(await _releaseFileRepository.Get(releaseId, file.Id)).Name ?? "Unknown" `